libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    public class DialogContext
+    {
+        private object _finalResult;
```

Comment:
Is this name right? From usage, it seems more like "_currentResult" would be a better name.

Reply:
The name is poor but `_currentResult` wouldn't be accurate.  It's the result returned by the dialog on the bottom of the stack when it ends and the stack is now empty.  This is needed to make prompts (and other controls) work in free standing mode.  Again I can get into why but I don't know that we have time and this is an implementation detail that no consumer will ever see.